apps/files/lib/Command/CheckCache.php

```
@@ -0,0 +1,103 @@
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CheckCache extends Command {
+	const ERROR_MISSING_FILE = 1;
+	const ERROR_NOT_A_FILE = 2;
+	const ERROR_CANNOT_OPEN = 3;
+
+	/** @var IRootFolder */
+	private $rootFolder;
+
+	public function __construct(IRootFolder $rootFolder) {
+		parent::__construct();
+		$this->rootFolder = $rootFolder;
+	}
+
+	protected function configure() {
+		$this->setName('files:check-cache')
+			->setDescription('Check if the target file exists in the primary storage')
+			->addArgument(
+				'uid',
+				InputArgument::REQUIRED,
+				'The user (user id) who owns the file'
+			)
+			->addArgument(
+				'target-file',
+				InputArgument::REQUIRED,
+				'The file we want to check'
+			)->addOption(
+				'remove',
+				null,
+				InputOption::VALUE_NONE,
+				'Remove the file from the cache if it\'s missing in the backend'
+			);
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$uid = $input->getArgument('uid');
+		$userFolder = $this->rootFolder->getUserFolder($uid);  // might throw a NoUserException
+
+		try {
+			$targetFile = $input->getArgument('target-file');
+			$node = $userFolder->get($targetFile);
+		} catch (NotFoundException $ex) {
+			$output->writeln("<error>$targetFile not found within user $uid</error>");
+			return self::ERROR_MISSING_FILE;
+		}
+
+		if ($node->isShared() || !($node->getStorage()->instanceOfStorage(IHomeStorage::class))) {
+			$output->writeln("<comment>Ignoring $targetFile because it is shared or not inside the primary storage</comment>");
+			return 0;
+		}
+
+		if (!($node instanceof File)) {
+			$output->writeln("<error>$targetFile isn't a file</error>");
+			return self::ERROR_NOT_A_FILE;
+		}
+
+		$stream = @$node->fopen('rb');  // errors are expected if the file is missing
```

Need to check if we want to suppress the errors or not. I assume the admin knows what file is giving problems as well as the reason behind those problems, so there is no reason to log the same thing again.
Note that, taking into account the current implementation of the objectstore, we can't know the reason why the file can't be opened. We're assuming here that the file is missing, but it might happen that there is no connection or wrong permissions; in this case, we rely on the admin to use or not this command knowing that the file (and/or the filecache infor) might be deleted.